nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/AbstractHiveQLProcessor.java

```diff
@@ -75,6 +81,38 @@
             .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
             .build();
 
+    public static final PropertyDescriptor QUERY_TIMEOUT = new PropertyDescriptor.Builder()
+            .name("hive-query-timeout")
+            .displayName("Query timeout")
+            .description("Sets the number of seconds the driver will wait for a query to execute. "
+                    + "A value of 0 means no timeout. NOTE: Non-zero values may not be supported by the driver.")
+            .defaultValue("0")
+            .required(true)
+            .addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        final List<ValidationResult> problems = new ArrayList<>(1);
+
+        if(validationContext.getProperty(QUERY_TIMEOUT).isSet()
+                && !validationContext.getProperty(QUERY_TIMEOUT).isExpressionLanguagePresent()
+                && validationContext.getProperty(QUERY_TIMEOUT).asInteger() != 0) {
+            try(HiveStatement stmt = new HiveStatement(null, null, null)) {
+                stmt.setQueryTimeout(0);
```

Comment:
You say in the description:

> A value of 0 means no timeout. NOTE: Non-zero values may not be supported by the driver.

Shouldn't the test be on a non-zero value here?

Reply:
Actually, in versions of the driver that does not implement this method, this call will throw an exception no matter what is the value.

Reply:
True (at least for older Apache Hive drivers), but it is a tad confusing to see the if statement check for non-zero then test with zero.

Also, I can't find the discussion but I thought we were going to do similar error handling in the setTimeout() method below as we do in the customValidate, for when Expression Language is present but the driver doesn't support non-zero values. IIRC it would allow a query timeout of zero if the driver didn't support it, but if the user set it to a positive value and the driver didn't support it, it would throw an error (akin to being invalid if found in customValidate())?

Reply:
Did you mean this one: https://github.com/apache/nifi/pull/2138#discussion_r159670913

Reply:
> True (at least for older Apache Hive drivers), but it is a tad confusing to see the if statement check for non-zero then test with zero.

Do you suggest an arbitrary value like
````java
stmt.setQueryTimeout(1); // just checking driver supports query timeout
````
?

Reply:
Yeah anything but zero, just so it matches the logic

Reply:
Also I think yes, the idea is to catch the "not supported" thing during customValidate() where we can, but we also have to add it to the runtime checking because of EL